static/app/components/core/alert/alert.tsx

```diff
@@ -21,6 +21,7 @@ export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
   type: 'muted' | 'info' | 'warning' | 'success' | 'error';
   defaultExpanded?: boolean;
   expand?: React.ReactNode;
+  handleExpandChange?: (isExpanded: boolean) => void;
```

Comment:
i elected to add a generic handler instead of modifying alert to work specifically for this use case (subset of expansion items on collapse, all expansion items on expand) nor creating the entirety of the expand behavior in `pendingChangea`